lib/src/main/java/com/sensirion/libble/services/sensirion/smartgadget/SmartgadgetTemperatureService.java

```diff
@@ -30,19 +38,72 @@ public SmartgadgetTemperatureService(@NonNull final Peripheral peripheral, @NonN
     }
 
     /**
+     * Checks if the incoming temperature value is feasible considering the device operating ranges
+     *
+     * @param temperatureInCelsius that will be checked
+     * @return <code>true</code> if the temperature value is feasible - <code>false</code> otherwise
+     */
+    private static boolean isTemperatureValueFeasible(@Nullable final Float temperatureInCelsius) {
```

Comment:
@abrauchli 
At the moment this approach do not notify unfeasible temperatures. I can change it to notify that temperatures with a NaN instead.

I can remove the unbox type if it is a problem.